interop-cats/jvm/src/main/scala/scalaz/zio/interop/stm/TRef.scala

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2017-2019 John A. De Goes and the ZIO Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package scalaz.zio.interop.stm
+
+import cats.~>
+import scalaz.zio.ZIO
+import scalaz.zio.stm.{ TRef => ZTRef }
+
+class TRef[F[+ _], A] private (underlying: ZTRef[A])(implicit liftIO: ZIO[Any, Throwable, ?] ~> F) {
```

Comment:
Same comment here for the implicit `FunctionK`. I'd delete that, make `TRef` extend `AnyVal`, and push the implicit `Runtime[Any]` to the `commit` method. I won't comment individually on these methods.

Reply:
fixed